### PR TITLE
Don't offer to use pattern matching if a user defined operator was involved.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/21551

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by CyrusNajmabadi** [comment]
retest this please

File: src/Features/CSharp/Portable/UsePatternMatching/CSharpAsAndNullCheckDiagnosticAnalyzer.cs [hunk 1]
```diff
@@ -95,6 +95,11 @@ private void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext)
             }
 
             var semanticModel = syntaxContext.SemanticModel;
+            if (semanticModel.GetSymbolInfo(comparison).GetAnySymbol().IsUserDefinedOperator())
```
Comment (Pilchie) [inline src/Features/CSharp/Portable/UsePatternMatching/CSharpAsAndNullCheckDiagnosticAnalyzer.cs]: Possible nulls?

Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/UsePatternMatching/CSharpAsAndNullCheckDiagnosticAnalyzer.cs]: GetSymbolInfo returns a struct, so GetAnySymbol is safe.  IsUserDefinedOperator is one of our 'Is' extensions, and as such is null-safe.

**Comment by CyrusNajmabadi** [comment]
retest windows_release_unit64_prtest  please

**Comment by CyrusNajmabadi** [comment]
retest windows_determinism_prtest please

**Pull request merged**